pkg/serving/v1alpha1/client_test.go

```diff
@@ -0,0 +1,320 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1/fake"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	client_testing "k8s.io/client-go/testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/knative/client/pkg/serving"
+)
+
+func TestClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "KnClient Tests")
+}
+
+var namespace = "test-ns"
+
+var _ = Describe("Client", func() {
+
+	var (
+		knClient KnClient
+		serving  fake.FakeServingV1alpha1
+	)
+
+	BeforeEach(func() {
+		serving = fake.FakeServingV1alpha1{Fake: &client_testing.Fake{}}
+		knClient = NewKnServingClient(&serving, namespace)
+	})
+
+	// -----------------------------------------------------------------------------------------
+	Describe("Getting a service", func() {
```

Comment:
One thing I like to do is to have a `Describe("Function")` for every public function or method that is being tested. This is especially useful is the test file matches a struct with lots of methods.

But as you can imagine. There are lots of way to organize and I tend not to be pedantic about things like that :)

Reply:
Yeah, just tried to be a bit more 'behavioral' to use some natural service. We could as well use 'GetService', as description name. I don't have a strong opinion here.